docs/scripts/getTypeScriptProjects.ts

```diff
@@ -0,0 +1,106 @@
+import path from 'path';
+import * as ts from 'typescript';
+import { Project, ProjectNames, Projects } from './api/utils';
+
+const workspaceRoot = path.resolve(__dirname, '../../');
+
+interface CreateProgramOptions {
+  name: ProjectNames;
+  rootPath: string;
+
+  /**
+   * Config to us to build this package
+   * The path must be relative to the root path.
+   */
+  tsConfigPath: string;
+
+  /**
+   * File used as root of the package
+   * The path must be relative to the root path.
+   */
+  entryPointPath: string;
+
+  /**
+   * Folder containing all the components of this package
+   * The path must be relative to the root path.
+   */
+  componentsFolder?: string;
+
+  /**
+   * Additional files containing components outside the component's folder
+   * The path must be relative to the root path.
+   */
+  otherComponentFiles?: string[];
+}
+
+const createProject = (options: CreateProgramOptions): Project => {
+  const { name, tsConfigPath, rootPath, entryPointPath, componentsFolder, otherComponentFiles } =
+    options;
+
+  const compilerOptions = ts.parseJsonConfigFileContent(
```

Comment:
The current generator uses this helper: https://github.com/mui/mui-x/blob/5f6b29d6d48753ffcb555c03df07cd18e4a90cc8/docs/scripts/generateProptypes.ts#L8

It parses and returns the compiler options, also checking if there's an error.

Reply:
I try not to use TTP in a code that is also called for `docs:api:build`
But the logic is very simple and can be inlined.